### PR TITLE
[skrifa] tthint: use cvar table

### PR DESCRIPTION
**Pull request by dfrg**
Applies cvar deltas to the cvt.

### CODE REVIEW

File: skrifa/src/outline/glyf/hint/instance.rs [hunk 1]
```diff
@@ -171,13 +171,40 @@ impl HintInstance {
             Definition::default(),
         );
         self.cvt.clear();
+        if let Some(cvar) = outlines.cvar.as_ref() {
```
Comment (cmyr) [inline skrifa/src/outline/glyf/hint/instance.rs:174]: not sure it's useful here, but something that I have found helpful for the idea of "I need both of these optional things to exist" is,

```rust
if let Some((one, two)) = opt1.zip(opt2) {
...
}
```

File: skrifa/src/outline/glyf/hint/instance.rs [hunk 2]
```diff
@@ -171,13 +171,40 @@ impl HintInstance {
...
+            if let Ok(var_data) = cvar.variation_data(axis_count) {
+                for (tuple, scalar) in var_data.active_tuples_at(coords) {
+                    for delta in tuple.deltas() {
+                        let ix = delta.position as usize;
+                        if let Some(value) = self.cvt.get_mut(ix) {
+                            *value += delta.apply_scalar(scalar).to_bits();
+                        }
+                    }
+                }
+            }
```
Comment (cmyr) [inline skrifa/src/outline/glyf/hint/instance.rs:186]: it's easy to imagine a little helper method that handles accumulating the deltas?

**Review by cmyr — APPROVED** [review]
👍

File: skrifa/src/outline/glyf/hint/instance.rs (hunk 1 above)
Comment (dfrg) [reply skrifa/src/outline/glyf/hint/instance.rs:174]: ah, hadn't seen this before. I don't see how I'd use it here but will definitely add it to the toolbox!

File: skrifa/src/outline/glyf/hint/instance.rs (hunk 2 above)
Comment (dfrg) [reply skrifa/src/outline/glyf/hint/instance.rs:186]: good call. I'll add this to the cvar table

**Pull request merged**